controllers/storagecluster/cephcluster.go

```diff
@@ -1198,3 +1211,34 @@ func getMonitoringClient() (*monitoringclient.Clientset, error) {
 	}
 	return client, nil
 }
+
+// getIPFamilyConfig checks for a Single Stack IPv6 or a Dual Stack cluster
+func getIPFamilyConfig(c client.Client) (rookCephv1.IPFamilyType, bool, error) {
+	isIPv6 := false
+	isIPv4 := false
+	networkConfig := &configv1.Network{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: "cluster", Namespace: ""}, networkConfig)
+	if err != nil {
+		return "", false, fmt.Errorf("could not get network config details. %v", err)
+	}
+
+	for _, cidr := range networkConfig.Status.ClusterNetwork {
+		if strings.Count(cidr.CIDR, ":") < 2 {
+			isIPv4 = true
+		} else if strings.Count(cidr.CIDR, ":") >= 2 {
+			isIPv6 = true
+		}
+	}
+
+	// Dual Stack is not supported in downstream ODF.
+	// So use IPv4 and DualStack:False in case of dual stack cluster
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1804290
+	// https://bugzilla.redhat.com/show_bug.cgi?id=2181350#c10
+	if isIPv4 && isIPv6 {
+		return rookCephv1.IPv4, false, nil
+	} else if isIPv6 && !isIPv4 { // IPv6 single stack cluster
+		return rookCephv1.IPv6, false, nil
+	}
```

Comment:
Can you pls add a comment why we are sending empty if it is ipv4, So that there are no confusions in the future.

Reply:
Agreed that it's confusing. I've updated the function to return IPFamily in case of single stack clusters.